bindata/network/ovn-kubernetes-ipsec/ovn-ipsec.yaml

```diff
@@ -0,0 +1,207 @@
+kind: DaemonSet
+apiVersion: apps/v1
+metadata:
+  name: ovn-ipsec
+  namespace: openshift-ovn-kubernetes
+  annotations:
+    kubernetes.io/description: |
+      This DaemonSet launches the ovn ipsec networking components for all nodes.
+    release.openshift.io/version: "{{.ReleaseVersion}}"
+spec:
+  selector:
+    matchLabels:
+      app: ovn-ipsec
+  updateStrategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        app: ovn-ipsec
+        component: network
+        type: infra
+        openshift.io/component: network
+        kubernetes.io/os: "linux"
+    spec:
+      serviceAccountName: ovn-kubernetes-node
+      hostNetwork: true
+      hostPID: true
+      priorityClassName: "system-node-critical"
+      # volumes in all containers:
+      # (container) -> (host)
+      # /etc/openvswitch -> /var/lib/openvswitch/etc - ovsdb system id
+      # /var/lib/openvswitch -> /var/lib/openvswitch/data - ovsdb data
+      # /run/openvswitch -> tmpfs - ovsdb sockets
+      # /env -> configmap env-overrides - debug overrides
+      containers:
+      # ovs-monitor-ipsec and libreswan daemons
+      - name: ovn-ipsec
+        image: "{{.OvnImage}}"
+        command:
+        - /bin/bash
+        - -c
+        - |
+          #!/bin/bash
+
+          function cleanup()
```

Comment:
This daemonset will be stopped on live nodes during upgrades - will this cause traffic to be dropped? (I assume so).

Reply:
In what contexts do we stop/kill the ovn-ipsec and libreswan containers? Is this during cert rotation? If so, cleaning out the xfrm policy and xfrm state on each node by node basis will cause replay errors on other nodes sending traffic to the node where the container is being re-provisioned?

Reply:
The way this is written, it will definitely blackhole the node during in-place upgrades - no good :-)

Reply:
I had to look into this more because now I have a working openshift cluster and I could do some testing and investigation. Currently, both libreswan and ovs-monitor-ipsec do not allow a clean restart without flushing flows from xfrm (directly by libreswan but indirectly by ovs-monitor-ipsec - which is unpleasant. This functionality is in progress in libreswan and easy to implement in ovs. We can uncleanly exit (i.e. kill) both and this should work but its not ideal.

I guess the other option is to move libreswan to the host and do what was done with OVS. In this way, a container would never restart libreswan. ovs-monitor-ipsec could be modified to exit without flushing anything. Not sure how much effort it is to move libreswan to the host?

Reply:
Looking at https://github.com/libreswan/libreswan/issues/385 and seeing what was done with OVS in CNO before it moved to the host, perhaps restarting ovs-monitor-ipsec and pluto ungracefully will be ok. When the libreswan and ovs-monitor-ipsec features get upstreamed, I can modify to a cleaner solution. What do you think?